gamemodes/terrortown/entities/entities/ttt_flame.lua

```diff
@@ -0,0 +1,259 @@
+	-- @realm server
+	function ENT:OnTakeDamage()
+	end
+
+	---
+	-- @param Vector pos The place a fire should originate from.
+	-- @param TraceResult tr The trace to set a fire with.
+	-- @param number num The number of fires to start.
+	-- @param number lifetime The base lifespan of each flame.
+	-- @param boolean explode Should the fire explode on death?
+	-- @param nil|Player dmgowner The player that began the fire, if any.
+	-- @realm server
+	function StartFires(pos, tr, num, lifetime, explode, dmgowner)
+		for i = 1, num do
+			local ang = Angle(-math.Rand(0, 180), math.Rand(0, 360), math.Rand(0, 360))
+
+			local vstart = pos + tr.HitNormal * 64
+
+			local flame = ents.Create("ttt_flame")
+			flame:SetPos(vstart)
+			if IsValid(dmgowner) and dmgowner:IsPlayer() then
+				flame:SetDamageParent(dmgowner)
+				flame:SetOwner(dmgowner)
+			end
+			flame:SetDieTime(CurTime() + lifetime + math.Rand(-2, 2))
+			flame:SetExplodeOnDeath(explode)
+
+			flame:Spawn()
+			flame:PhysWake()
+
+			local phys = flame:GetPhysicsObject()
+			if IsValid(phys) then
+				-- the balance between mass and force is subtle, be careful adjusting
+				phys:SetMass(2)
+				phys:ApplyForceCenter(ang:Forward() * 500)
+				phys:AddAngleVelocity(Vector(ang.p, ang.r, ang.y))
+			end
+		end
+	end
+
+	---
+	-- @param Vector pos The place a fire should originate from.
+	-- @param number size How large a fire should be.
+	-- @param number attack How much damage a fire should inflict.
+	-- @param number fuel The health / duration of a fire.
+	-- @param nil|Player dmgowner The player that began the fire, if any.
+	-- @param nil|Entity parent The entity that controls the created fire, usually a ttt_flame.
+	-- @realm server
+	function SpawnFire(pos, size, attack, fuel, owner, parent)
+		local fire = ents.Create("env_fire")
+		if not IsValid(fire) then return end
+
+		fire:SetParent(parent)
+		fire:SetOwner(owner)
+		fire:SetPos(pos)
+		--no glow + delete when out + start on + last forever
+		fire:SetKeyValue("spawnflags", tostring(128 + 32 + 4 + 2 + 1))
+		fire:SetKeyValue("firesize", size * math.Rand(0.7, 1.1))
+		fire:SetKeyValue("fireattack", attack)
+		fire:SetKeyValue("health", fuel)
+		fire:SetKeyValue("damagescale", "-10") -- only neg. value prevents dmg
+
+		fire:Spawn()
+		fire:Activate()
+
+		return fire
+	end
+
+	---
+	-- greatly simplified version of SDK's game_shard/gamerules.cpp:RadiusDamage
+	-- does no block checking, radius should be very small
+	-- @param DamageInfo dmginfo The damage to inflict.
+	-- @param Vector pos The origin of the damage radius.
+	-- @param number radius The radius of the damage to inflict.
+	-- @param nil|Player inflictor The player responsible for the damage.
+	-- @realm server
+	function RadiusDamage(dmginfo, pos, radius, inflictor)
+		for k, vic in ipairs(ents.FindInSphere(pos, radius)) do
+			if IsValid(vic) and inflictor:Visible(vic) and vic:IsPlayer() and vic:Alive() and vic:IsTerror() then
+				vic:TakeDamageInfo(dmginfo)
+			end
```

Comment:
```suggestion
			if not IsValid(vic) or not inflictor:Visible(vic) or not vic:IsPlayer() or not vic:IsTerror() then continue end

			vic:TakeDamageInfo(dmginfo)
```

I'm pretty sure the `Alive` check is not needed, `IsTerror` means alive in the TTT sense. Even a spectator returns true in `Alive`